tests/nlu/data_augmentation/test_augmenter.py

```diff
@@ -37,167 +35,202 @@ def test_augmenter_create_tokenizer(run: Callable[..., RunResult],):
-    from rasa.nlu.data_augmentation.augmenter import _create_paraphrase_pool
 
-    data_root = os.path.dirname(
-        os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+    assert comparator_fn_1(
+        len(intents_to_augment), num_intents_to_augment_per_criterion
     )
+    assert comparator_fn_2(len(intents_to_augment), num_intents)
+
+
+@pytest.mark.parametrize(
+    "intent_to_augment, min_paraphrase_sim_score, max_paraphrase_sim_score, expected_len",
+    [
+        ("ask_transfer_charge", 0.98, 0.99, 0),
+        ("check_earnings", 0.98, 0.99, 0),
+        ("check_recipients", 0.98, 0.99, 0),
+        ("ask_transfer_charge", 0.0, 0.99, 6),
+        ("check_earnings", 0.0, 0.99, 18),
+        ("check_recipients", 0.0, 0.99, 18),
+        ("ask_transfer_charge", 0.0, 0.45, 2),
+        ("check_earnings", 0.0, 0.45, 3),
+        ("check_recipients", 0.0, 0.45, 8),
+        ("ask_transfer_charge", 0.4, 0.5, 1),
+        ("check_earnings", 0.4, 0.5, 5),
+        ("check_recipients", 0.4, 0.5, 8),
+    ],
+)
+def test_augmenter_paraphrase_pool_creation(
+    intent_to_augment: Text,
+    min_paraphrase_sim_score: float,
+    max_paraphrase_sim_score: float,
+    expected_len: int,
+):
+    from rasa.nlu.data_augmentation.augmenter import _create_paraphrase_pool
 
     paraphrases = rasa.shared.nlu.training_data.loading.load_data(
-        os.path.join(data_root, "data/test_nlu_paraphrasing/paraphrases.yml")
+        "data/test_nlu_paraphrasing/paraphrases.yml"
     )
 
-    # See the file `rasa/data/test_nlu_paraphrasing/nlu_train.yml` for available intents
-    intents_to_augment = {"ask_transfer_charge", "check_earnings", "check_recipients"}
-
-    # Max sim score in `rasa/data/test_nlu_paraphrasing/paraphrases.yml` is 0.975
-    # Min sim score in `rasa/data/test_nlu_paraphrasing/paraphrases.yml` is 0.389
-
     pool = _create_paraphrase_pool(
         paraphrases=paraphrases,
-        intents_to_augment=intents_to_augment,
-        min_paraphrase_sim_score=0.98,
-        max_paraphrase_sim_score=0.99,
+        intents_to_augment={intent_to_augment},
+        min_paraphrase_sim_score=min_paraphrase_sim_score,
+        max_paraphrase_sim_score=max_paraphrase_sim_score,
     )
-    assert len(pool) == 0
+    assert len(pool.get(intent_to_augment, [])) == expected_len
+
+
+@pytest.mark.parametrize(
+    "intents_to_augment",
+    [
+        set(),
+        {"check_recipients", "check_earnings"},
+        {
+            "ask_transfer_charge",
+            "greet",
+            "help",
+            "deny",
+            "check_recipients",
+            "thankyou",
+            "search_transactions",
+            "check_balance",
+            "pay_cc",
+            "goodbye",
+            "affirm",
+            "inform",
+            "check_earnings",
+            "human_handoff",
+            "transfer_money",
+        },
+    ],
+)
+def test_augmenter_paraphrase_pool_structure(intents_to_augment: Set[Text]):
```

Comment:
Do we need a separate test? If we are only checking if the number of intents in the returned pool are equal to the number of intents in `intents_to_augment` then we can probably change the parameterization of `test_augmenter_paraphrase_pool_creation` to include this there itself?

Reply:
idk, to me they serve slightly different purposes, i.e. `test_augmenter_paraphrase_pool_creation` assess whether the right number of examples for a given intent are returned w.r.t. the thresholds and `test_augmeter_paraphrase_pool_structure` tests that if we get data for all the intents (ignoring scores).

I'll probably rename them to reflect their different purposes - wdyt?

Reply:
Okay, let's rename them in that case.